qiskit_experiments/library/characterization/t2ramsey.py

```diff
@@ -94,14 +95,33 @@ def __init__(
             unit: Optional, time unit of `delays`.
                 Supported units: 's', 'ms', 'us', 'ns', 'ps', 'dt'. The unit is
                 used for both T2Ramsey and for the frequency.
-            osc_freq: the oscillation frequency induced by the user. \
-            The frequency is given in Hz.
+            osc_freq: the oscillation frequency induced by the user.
+                The frequency is given in Hz.
 
         """
 
         super().__init__([qubit])
         self.set_experiment_options(delays=delays, unit=unit, osc_freq=osc_freq)
 
+    def set_experiment_options(self, **fields):
+        """Set the experiment options.
+
+        Args:
+            fields: The fields to update the options
+
+        Raises:
+            AttributeError: If the field passed in is not a supported options
+        """
+        super().set_experiment_options(**fields)
+
+        # set frequency guess from experiment configuration
+        if "osc_freq" in fields:
```

Comment:
T2Ramsey uses user input frequency offset as an initial guess of frequency. However, this is only valid when the offset is enough larger than the frequency error of drive pulse from the true 0-1 transition energy. Indeed this is not usually guaranteed.

The guess algorithm newly implemented in `DumpedOscillationAnalysis` is enough smart and can provide enough accurate guess even when the sampling range is smaller than one cycle of oscillation. Considering that fact, we can remove this logic.

Reply:
Is there an advantage in reducing the sampling range? Because I would expect both T2* and the frequency to be less accurate in this case. Another question is what does the user expect. In Ignis, the user provided the `osc_freq`.